app/src/main/java/com/thinqtv/thinqtv_android/MainActivity.java

```diff
@@ -66,4 +90,252 @@ public void goGetInvolved(View V){
+                constraintLayout.addView(newEvent_name);
+                constraintLayout.addView(newEvent_host);
+                constraintLayout.addView(newEvent_time);
+
+                // Add simple divider to put in between ConstraintLayouts (ie events)
+                View viewDivider = new View(this);
+                viewDivider.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, 2));
+                viewDivider.setBackgroundColor(Color.LTGRAY);
+
+                // Get the selected event filter text
+                Spinner spinner = (Spinner)findViewById(R.id.eventsSpinner);
+                String text = spinner.getSelectedItem().toString();
+
+                //get current date and what week it is
+                Calendar mCalendar = Calendar.getInstance();
+                int week = mCalendar.get(Calendar.WEEK_OF_MONTH);
+
+                switch(text)
+                {
+                    case ("All Events") :
+                    {
+                        linearLayout.addView(constraintLayout);
+                        linearLayout.addView(viewDivider);
+                        break;
+                    }
+                    case ("This Week") :
+                    {
+                        mCalendar.set(Calendar.WEEK_OF_MONTH, ++week);
+                        Date filterDate = mCalendar.getTime();
+
+                        if (date.before(filterDate))
+                        {
+                            linearLayout.addView(constraintLayout);
+                            linearLayout.addView(viewDivider);
+                        }
+                        break;
+                    }
+                    case ("Next Week") :
+                    {
+                        mCalendar.set(Calendar.WEEK_OF_MONTH, (week + 1));
+                        Date filterDate = mCalendar.getTime();
+
+                        if (date.after(filterDate))
+                        {
+                            mCalendar.set(Calendar.WEEK_OF_MONTH, (week + 2));
+                            filterDate = mCalendar.getTime();
+
+                            if (date.before(filterDate))
+                            {
+                                linearLayout.addView(constraintLayout);
+                                linearLayout.addView(viewDivider);
+                            }
+                        }
+                        break;
+                    }
+                    case ("Future") :
+                    {
+                        mCalendar.set(Calendar.WEEK_OF_MONTH, (week + 2));
+                        Date filterDate = mCalendar.getTime();
+
+                        if (date.after(filterDate))
+                        {
+                            linearLayout.addView(constraintLayout);
+                            linearLayout.addView(viewDivider);
+                        }
+                        break;
+                    }
+                }
+            }
```

Comment:
Can you just double check that nothing in this for loop can be moved to before the for loop? Like for example getting the current date - I don't think we need to do that separately for each event, right?